core/src/main/scala/io/finch/response/package.scala

```diff
@@ -151,25 +153,39 @@ package object response {
    * An abstraction that is responsible for encoding the response of type `A`.
    */
   trait EncodeResponse[-A] {
-    def apply(rep: A): String
+    def apply(rep: A): Buf
     def contentType: String
+    def charset: Option[String] = Some("utf-8")
   }
 
   object EncodeResponse {
     /**
      * Convenience method for creating new [[io.finch.response.EncodeResponse EncodeResponse]] instances.
      */
-    def apply[A](ct: String)(fn: A => String): EncodeResponse[A] = new EncodeResponse[A] {
-      def apply(rep: A): String = fn(rep)
-      def contentType: String = ct
-    }
+    def apply[A](ct: String, cs: Option[String] = Some("utf-8"))(fn: A => Buf): EncodeResponse[A] =
+      new EncodeResponse[A] {
+        def apply(rep: A): Buf = fn(rep)
+        def contentType: String = ct
+        override def charset: Option[String] = cs
```

Comment:
For consistency, can we either add `override` to all the methods or remove it from `charset`?